### ISSUE TITLE
Tile fetching issue when maximum cache size is reached

### ISSUE THREAD

**Issue by ahocevar** [openlayers/ol3#4961]
#4805 reveals an issue with tile fetching. It can be easily reproduced by e.g. using the simple.html example and changing the code to:
```js
var map = new ol.Map({
  layers: [
    new ol.layer.Tile({
      source: new ol.source.OSM({cacheSize: 16})
    })
  ],
  target: 'map',
  view: new ol.View({
    center: [0, 0],
    zoom: 2
  })
});
```
After zooming in and out using wheel zoom a few times, no more tiles will be loaded and only a single tile will remain.

**Comment by ahocevar** [comment]
@fredj, do you have any idea what might be causing this?

**Comment by fredj** [comment]
It works if you zoom after all tile have been loaded

**Comment by fredj** [comment]
https://github.com/openlayers/ol3/blob/master/src/ol/map.js#L1035

The test is false and no more tiles are loaded

**Comment by ahocevar** [comment]
Great find @fredj. Are you able to code up a fix?

**Comment by ahocevar** [comment]
And thanks for the investigation!

**Comment by fredj** [comment]
tiles are unlisten before they are loaded; the tileCache is not notified and the tile is never removed from the list.
I will confirm this tomorrow and prepare a fix

**Comment by ahocevar** [comment]
Ideally, tile fetching would also work without a tile cache at all. Strange that the tile cache affects how the tile queue works. Thanks a lot for looking into this @fredj.

**Comment by fredj** [comment]
fixed with #5051
